packages/synthetix-main/contracts/modules/core/AccountModule.sol

```diff
@@ -28,6 +29,22 @@ contract AccountModule is IAccountModule, OwnableMixin, AccountRBACMixin, Associ
         return _getNft(_ACCOUNT_SYSTEM);
     }
 
+    function getAccountPermissions(uint accountId) external view returns (AccountPermission[] memory) {
+        AccountRBAC storage accountRbac = _accountModuleStore().accountsRBAC[accountId];
+
+        uint allPermissionsLength = accountRbac.permissionAddresses.length();
+        AccountPermission[] memory allPermissions = new AccountPermission[](allPermissionsLength);
+        for (uint i = 0; i < allPermissionsLength; i++) {
+            address permissionAddress = accountRbac.permissionAddresses.valueAt(i + 1);
+            allPermissions[i] = AccountPermission({
```

Comment:
Is it really necessary to create a struct in memory? Can't we just return a tuple with arrays?

Reply:
instead of having two arrays; having an array of objects is easier to consume on the FE.  i can change this, let me know if you feel strongly about this (also a rec from db to create a struct)